icu4j/main/core/src/test/java/com/ibm/icu/dev/test/rbbi/RBBITest.java

```diff
@@ -1002,6 +1002,28 @@ public void TestBug22602() {
             fail("TestBug22602: Unexpected exception while new RuleBasedBreakIterator() with a long string followed by a ';': " + e);
         }
     }
+    @Test
+    public void TestBug22636() {
+        try {
+            RuleBasedBreakIterator bi = new RuleBasedBreakIterator("A{77777777777777};");
+        }
+        catch (IllegalArgumentException e) {
+            // expected exception with a long string followed by a ';'.
+        }
+        catch(StackOverflowError e) {
```

Comment:
I don't think we need to check for StackOverflowError here.
Looks like it got here from a copy/paste from TestBug22602, which was throwing an unexpected StackOverflow, but that was never the situation with these new test cases for int overflow problems.